coalib/testing/LocalBearTestHelper.py

```diff
@@ -12,16 +13,38 @@
 
 
 @contextmanager
-def execute_bear(bear, *args, **kwargs):
+def execute_bear(bear, *args, **kwargs):  # pragma: no cover
     try:
-        bear_output_generator = bear.execute(*args, **kwargs)
+        console_output = []
+
+        with ExitStack() as stack:
+            if hasattr(bear, 'process_output'):
+                console_output.append('The program yielded '
+                                      'the following output:\n')
+                old_process_output = bear.process_output
+
+                def new_process_output(output, filename=None, file=None):
+                    if isinstance(output, tuple):
+                        stdout, stderr = output
+                        console_output.append('Stdout:\n' + stdout)
+                        console_output.append('Stderr:\n' + stderr)
+                    else:
+                        console_output.append(output)
+                    return old_process_output(output, filename, file)
+
+                stack.enter_context(patch.object(
+                    bear, 'process_output', wraps=new_process_output))
+
+            bear_output_generator = bear.execute(*args, **kwargs)
+
         assert bear_output_generator is not None, \
             'Bear returned None on execution\n'
         yield bear_output_generator
     except Exception as err:
         msg = []
         while not bear.message_queue.empty():
             msg.append(bear.message_queue.get().message)
+        msg.extend(console_output)
```

Comment:
@yash-nisar maybe you should really do `msg += console_output`, this is kinda easier to read :+1: